daemon/main.go

```diff
@@ -160,10 +160,19 @@ func getKernelVersion() (*go_version.Version, error) {
 		log.WithError(err).Fatal("kernel version: NOT OK")
 	}
 	verStrs := strings.Split(string(verOut), ".")
-	if len(verStrs) < 2 {
+	if len(verStrs) < 3 {
 		return nil, fmt.Errorf("unable to get kernel version from %q", string(verOut))
 	}
-	return go_version.NewVersion(strings.Join(verStrs[:2], "."))
+	// We are assuming the kernel version will be something as:
+	// 4.9.17-040917-generic
+
+	// If verStrs is []string{ "4", "9", "17-040917-generic" }
+	// then we need to split the patch string to get a number.
+	patchVersion := strings.Split(verStrs[2], "-")
+	if len(patchVersion) != 0 {
```

Comment:
Do we need to clear the third element of verStr when len(patchVersion) == 0? Otherwise we would just recombine the version including the full patch string.

Reply:
I've rewrote this function again